components/raftstore/src/store/fsm/peer.rs

```diff
@@ -4163,20 +4169,40 @@ where
                 "err" => %e,
             );
         }
-        if let Err(e) = self.fsm.peer.destroy(
+
+        // Here, drop the lock of `StoreMeta` to make the preparations for destroy
+        // safety to be executed.
```

Comment:
Is this the intended meaning?
```suggestion
        // Here, drop the lock of `StoreMeta` to avoid blocking other threads.
```